src/main/java/com/nanalysis/sharepoint/SharepointClient.java

```diff
@@ -224,6 +224,9 @@ public InputStream download(String folder, String filename) throws IOException,
                 .build();
 
         HttpResponse<InputStream> response = httpClient.send(request, HttpResponse.BodyHandlers.ofInputStream());
+        if(response.statusCode() != 200) {
+            throw new IOException("Unable to download file: HTTP error " + response.statusCode());
+        }
```

Comment:
No, that's a HTTP status code. Ideally I would have preferred to use a helper function, because all 20x and 30x can be considered some kind of success, and all 40x and 50x are errors. 30x are for things like redirections.

200 is the normal success code, I hope sharepoint doesn't try to be smart and reply with anything else in the future.
What I really wanted to avoid was to continue as if everything was fine when the file was missing.